### DISCUSSION TITLE
[Question]: How to Expand/Collapse Right Panel (Header Panel)

### DISCUSSION THREAD

**Discussion by AntonSRial** [carbon-design-system/carbon#11612] [Help] [answered]
### Question for Carbon

Hi!

I'm trying to implement the `Right Panel` in my app but I can't find a way to collapse/expand this component. According to documentation, the panel is attached to a `HeaderGlobalAction` and it's visibility is controlled by the prop `expanded`. However, it's no example avaliable in the wiki of how to toggle the expand/collapse action.

Could you give me some advice? Thanks in regard.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by jnm2377** [comment, marked as answer]
Hi @AntonSRial to collapse and expand the `HeaderPanel` component, you'd need to use a state. Set the state value as the `expanded` prop value, and on `HeaderGlobalAction` click you would update that state to toggle the expansion.

Something like:
```jsx
const [isExpanded, setIsExpanded] = useState(false);

const toggleExpand = ( ) => { 
setIsExpanded(!isExpanded)
}

<HeaderGlobalAction onClick={toggleExpand}> [icon] </HeaderGlobalAction>
<HeaderPanel expanded={isExpanded}> [panel items] </HeaderPanel>

```
